tests/test_sliceable_buffer.py

```diff
@@ -0,0 +1,200 @@
+from typing import NamedTuple, Sequence, Callable
+
+from _pytest.python_api import raises
+
+from amazon.ion.sliceable_buffer import IncompleteReadError, SliceableBuffer
+from tests import parametrize
+
+"""
+This test rips off the pattern from test_reader_buffer and the relevant
+(for now) tests.
+
+The SliceableBuffer is not intended as a drop-in replacement for the
+BufferQueue so there are just enough differences to warrant copying and 
+modifying vs re-using as-is.
+
+The intention/hope is that eventually that buffer will be replaced by this
+one and then this will be the only remaining test.
+"""
+
+
+def read(expected):
+    def action(buffer):
+        read_len = len(expected)
+        (actual, buffer) = buffer.read_slice(read_len)
+        assert actual == expected
+
+        return -read_len, buffer
+
+    return action
+
+
+def expect(exception, action):
+    def raises_action(buffer):
+        with raises(exception):
+            action(buffer)
+        return 0, buffer
+
+    return raises_action
+
+
+def extend(data):
+    def action(buffer):
+        return len(data), buffer.extend(data)
+
+    return action
+
+
+def read_byte(expected):
+    def action(buffer):
+        (actual, buffer) = buffer.read_byte()
+        assert expected[0] == actual
+
+        return -1, buffer
+
+    return action
+
+
+def skip(n, expected_rem=0):
+    def action(buffer):
+        (skipped, buffer) = buffer.skip(n)
+        assert expected_rem == n - skipped
+        return -skipped, buffer
+
+    return action
+
+
+class _P(NamedTuple):
+    desc: str
+    actions: Sequence[Callable]
+    is_unicode: bool = False
+
+    def __str__(self):
+        return self.desc
+
+
+@parametrize(
```

Comment:
Let's add some parameters to test the expected `IncompleteReadError` behavior.

Reply:
I think the `read_byte` as already covered... there's nothing really interesting there as the buffer is either empty or it's not. I added tests to better verify lack of mutation for `read_slice` and resulting buffer for partial skip.